lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
-    fieldSpec.typeOptions = {};
-  }
 
-  if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
-    throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
-  }
+    if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
+      throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
+    }
 
-  returnedTypeOptions = this.types[fieldSpec.type].validateFieldSpecification(fieldSpec.typeOptions);
+    const returnedTypeOptions = this.types[fieldSpec.type].validateFieldSpecification(fieldSpec.typeOptions);
 
-  if (typeof returnedTypeOptions === 'boolean') {
-    if (!returnedTypeOptions) {
+    if (typeof returnedTypeOptions === 'boolean' && !returnedTypeOptions) {
       throwOrStoreError(`Field of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
     }
-  }
-
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
-  }
-
-  fieldSpec.typeOptions = returnedTypeOptions;
-  return {isValid: true, fieldSpec: fieldSpec};
-};
-
-/**
- * @param {FieldSpecification} fieldSpec
- * @param {string} indexName
- * @param {string} collectionName
- * @param {string} fieldName
- * @param {boolean} verboseErrors
- * @returns {object}
- * @throws InternalError
- */
-Validation.prototype.curateFieldSpecificationFormat = function validationCurateFieldSpecificationFormat (fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
-  var errors = [];
 
-  verboseErrors = verboseErrors || false;
+    if (errors.length > 0) {
+      return {errors, isValid: false};
+    }
 
-  if (!checkAllowedProperties(fieldSpec, ['mandatory', 'type', 'defaultValue', 'description', 'multivalued', 'typeOptions'])) {
-    throwOrStoreError(`The field ${indexName}.${collectionName}.${fieldName} specification has invalid properties.`, verboseErrors, errors);
+    fieldSpec.typeOptions = returnedTypeOptions;
+    return {fieldSpec, isValid: true};
   }
 
-  mandatoryFieldSpecProperties.forEach(propertyName => {
-    if (!fieldSpec.hasOwnProperty(propertyName)) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, ${propertyName} is a mandatory field specification property.`, verboseErrors, errors);
+  /**
+   * @param {FieldSpecification} fieldSpec
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {string} fieldName
+   * @param {boolean} [verboseErrors]
+   * @returns {object}
+   * @throws InternalError
+   */
+  curateFieldSpecificationFormat(fieldSpec, indexName, collectionName, fieldName, verboseErrors = false) {
+    const errors = [];
+
+    if (!checkAllowedProperties(fieldSpec, ['mandatory', 'type', 'defaultValue', 'description', 'multivalued', 'typeOptions'])) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 129 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)